Add documentation for noaaweather sensor integration

rewsssiestedbo opened the pull request:
**Description:**


**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#23875

## Checklist:

- [X] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [X] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

rewsssiestedbo committed:
Add documentation for noaaweather sensor component

klaasnicolaas committed:
:pencil2: Tweak

klaasnicolaas committed:
:pencil2: Tweak

klaasnicolaas commented on source/_components/noaaweather.markdown:
```diff
@@ -0,0 +1,116 @@
+# Example configuration.yaml entry
+sensor:
+  - platform: noaaweather
+    useragent:	email@example.com
```
I think that we can remove the `useragent`, only required items in the first example.

klaasnicolaas commented on source/_components/noaaweather.markdown:
```diff
@@ -0,0 +1,116 @@
+comments: false
+sharing: true
+footer: true
+logo: 
```
I have added already the variable for you, but could you add a logo of the integration?

MatthewFlamm replied:
This integration will use the same logo(s) as the one in [#9372](https://github.com/home-assistant/home-assistant.io/pull/9372).  The same comment will apply that we need to request permission.  We are not using unmodified data products, both this platform/integration and the one in #9372 use minimally altered, but not unaltered data.

MatthewFlamm replied:
I retract this, see comment in #9372

rewsssiestedbo committed:
Update noaaweather.markdown

klaasnicolaas committed:
:pencil2: Tweak

MatthewFlamm commented:
Similar to comment in the backend PR, this one should be merged with [#9372](https://github.com/home-assistant/home-assistant.io/pull/9372) as they are the same data source, but one for sensor and one for weather platform.

rewsssiestedbo committed:
Remove usehaweather option.

The option was removed from the code, since it was unlikely to be used and worked differently from the approach in the corresponding nws weather module.

klaasnicolaas commented on source/_components/noaaweather.markdown:
```diff
@@ -0,0 +1,111 @@
+logo: noaa.png
+ha_category:
+  - Sensor
+ha_release: 0.94
```
```suggestion
ha_release: 0.96
```
